frontend/src/app/user-interface/burger-menu/burger-menu.component.ts

```diff
@@ -0,0 +1,38 @@
+import { Component, OnInit } from '@angular/core';
+import {MatMenuModule, MatButtonModule, MatIconModule} from '@angular/material';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+
+@Component({
+  selector: 'app-burger-menu',
+  templateUrl: './burger-menu.component.html',
+  styleUrls: ['./burger-menu.component.scss']
+})
+export class BurgerMenuComponent implements OnInit {
+  lastSelected:string = "";
+  constructor() { }
+
+  ngOnInit() {
+  }
+    onClick(event: Event){
+        let elementClass;
+        if(event.srcElement.children.length != 0)
+            elementClass = event.srcElement.id.toString().replace(/button_/g,'');
+        else
+            elementClass = ( <HTMLElement>event.target ).parentElement.id.toString().replace(/button_/g,'');
+        console.log(this.lastSelected);
+        if(this.lastSelected)
+        {
+            var prevElementList = document.querySelectorAll("[id^=" + CSS.escape(this.lastSelected) + "] ");
+            prevElementList.forEach ((item, index) => {
+                (document.getElementById(item.id) as HTMLElement).style.opacity="0";
```

Comment:
Czy linia 25 nie daje nam dostępu do wszystkich elementów? 27 Wydaje się być zbędna pod względem szukania elementów.
Wystarczy wykorzystać listę z linii 25 zamiast szukać w html jeszcze raz tych elementów
``` ts
      prevElementList.forEach(item => {
        (item as HTMLElement).style.opacity = '0';
      });
```
Co do `style.opacity = 0` raczej zrobiłbym na klasie np `highlighted`, jeżeli dojdzie nam funkcjonalność szukania to może pojawić się konflikt.

Reply:
poprawione, style nie dodane na razie